src/main/java/com/limechain/sync/fullsync/FullSyncMachine.java

```diff
@@ -13,6 +35,94 @@ public FullSyncMachine(final Network networkService) {
     }
 
     public void start() {
-        networkService.sendNeighbourMessages();
+        // TODO: DIRTY INITIALIZATION FIX:
+        //  this.networkService.currentSelectedPeer is null,
+        //  unless explicitly set via some of the "update..." methods
+        this.networkService.updateCurrentSelectedPeerWithNextBootnode();
+        AccessorHolder.getInstance().setToGenesis(); //todo: dirty fix for now
+
+        int startNumber = 1;
+        List<SyncMessage.BlockData> receivedBlockDatas = getBlocks(startNumber, 10);
+        while (!receivedBlockDatas.isEmpty()) {
+            startNumber += 10;
+            executeBlocks(receivedBlockDatas);
+            receivedBlockDatas = getBlocks(startNumber, 10);
+        }
```

Comment:
How was the number 10 chosen?

Reply:
It was chosen while testing, currently set to 50, not sure what would be the most appropriate number